### DISCUSSION TITLE
How to add a toolbar filter on Vue Admin Table

### DISCUSSION THREAD

**Discussion by aodihis** [craftcms/cms#13389] [Q&A]
Hello, 
I have a module that show a data using VueAdminTable, I want to know if is it possible to add the toolbar something like status dropdwon in Elemenet index page to do a filtering?

If it is possible, is there any documentation that I can follow?

Thank you.

**Comment by Anubarak** [comment]
Unfortunately there is no good way to do this without a page-reload or a destruction of the component.
I required it too and ultimately ended up creating a custom package in order to include a way to customize filters and and custom criteria parameters

**Comment by nfourtythree** [comment]
Hi @aodihis 

Yes, this is possible, there are actually examples of this in the Craft Commerce. For example the [shipping categories index template](https://github.com/craftcms/commerce/blob/develop/src/templates/shipping/shippingcategories/index.twig#L50-L74)

It uses the `actions` property on the `VueAdmintTable` class. You can find documentation on [action buttons](https://github.com/craftcms/cms/blob/develop/src/web/assets/admintable/README.md#action-buttons) in the admin table asset.

There is also further information about [buttons](https://github.com/craftcms/cms/blob/develop/src/web/assets/admintable/README.md#buttons) for creating a "New item" type button. You can enable a search toolbar for the table if you are retrieving data from an endpoint by settings the `search` property to `true`.

Hope this helps, thanks!

**Reply by nfourtythree** [reply]
Apologies if I have misunderstood the question, the information in this answer is for creating actions for example mass updating item statuses.

**Reply by Anubarak** [reply]
Do you know if there are any plans to include events/callbacks to manipulate query parameters when a reload is triggered?
What I actually did was to check if the api endpoint is typeof callable (so no string) in order to pass a callback function that will re-create the endpoint URL every time. That made it possible to change query parameters.

With this event it's much easier for others to inject a custom status menu via JS after the component is mounted and "reload" it on change along with the newly selected option.

**Reply by aodihis** [reply]
@nfourtythree sure no worries.

**Reply by nfourtythree** [reply]
Thank you for the feedback. Will take this internally and see what potential options we have for expanding the functionality. There definitely seems like there is something in this to make it more flexible and extensible. 

Thanks!

**Reply by janhenckens** [reply]
I'd like to add my vote for extending this :) I've been using the VueAdminTable in a couple of my plugins and in projects at the office, it's been fun to work with and it makes the custom things we build feel even more integrated with the rest of the CP, so I'm all for adding features like filters and callbacks 👍 The fact that this has docs helps too off course :)

**Reply by nfourtythree** [reply]
To keep everyone up to date, I have created a PR to the Craft repo to add some new features to the Vue admin table that should hopefully give the extensibility required.

Here is the PR https://github.com/craftcms/cms/pull/13416 in which there are examples of what new features are being added.

Thanks!

**Reply by janhenckens** [reply]
Thanks for keeping us posted @nfourtythree ❤️!

**Reply by brandonkelly** [reply]
Craft 4.5.0 is out with those changes.
